### PR TITLE
feat: adds images under manifest

### LINKED ISSUE #7181: List images associated to main manifest image in UI.
### Is your enhancement related to a problem? Please describe

There are a lot of <none> images which are actually associated to the manifest. But unfortunately within the Podman Desktop UI, it just says <none>.

### Describe the solution you'd like

We should have them "underneath" the image similar to how we showcase Compose / grouped pods.

### Describe alternatives you've considered

_No response_

### Additional context

Part of epic https://github.com/containers/podman-desktop/issues/6526

### PR DESCRIPTION
**Pull request by cdrage**
feat: adds images under manifest

### What does this PR do?

* Adds "children" images under the main manifest

### Screenshot / video of UI

Screenshot 2024-06-10 at 11 43 55 AM

https://github.com/containers/podman-desktop/assets/6422176/0077cb19-7cf2-4c0f-bce8-ee020b243907

### What issues does this PR fix or reference?

Closes https://github.com/containers/podman-desktop/issues/7181

### How to test this PR?

- [X] Tests are covering the bug fix or the new feature

1. Create a manifest
2. No longer see the <none> images and it will instead be under the
   manifest image.

Signed-off-by: Charlie Drage <charlie@charliedrage.com>

### CODE REVIEW

**Comment by cdrage** [comment]
I have created this WIP to show issue #7553 

For some reason I am getting odd issues with regards to selection:


https://github.com/containers/podman-desktop/assets/6422176/e02bc4fd-98df-436b-ab06-25e9a6fd7c96

@deboer-tim

**Comment by deboer-tim** [comment]
> For some reason I am getting odd issues with regards to selection:

Please try with the fix in #7555.

**Comment by cdrage** [comment]
> > For some reason I am getting odd issues with regards to selection:
> 
> Please try with the fix in #7555.

This fixes it thank you!

PR is also now ready for review 👍

**Comment by jeffmaury** [comment]
Can't make it to work.
I create a manifest using CLI:
```

{
    "schemaVersion": 2,
    "mediaType": "application/vnd.oci.image.index.v1+json",
    "manifests": [
        {
            "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
            "size": 926,
            "digest": "sha256:6a8487644ee8e5d4aac774380695b4dfc986a621a54f3d7a2f98c1dc6ad4a3a5",
            "platform": {
                "architecture": "arm64",
                "os": "linux"
            }
        },
        {
            "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
            "size": 595,
            "digest": "sha256:f711512233a66104534cf47880c2426dd9796b63ff12d733b0a94c6064b1cfb1",
            "platform": {
                "architecture": "amd64",
                "os": "linux"
            }
        }
    ]
}
```

But when I go the the images page, I have this:

image

**Comment by cdrage** [comment]
> Can't make it to work.
> 
> I create a manifest using CLI:
> 
> ```
> 
> 
> 
> {
> 
>     "schemaVersion": 2,
> 
>     "mediaType": "application/vnd.oci.image.index.v1+json",
> 
>     "manifests": [
> 
>         {
> 
>             "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
> 
>             "size": 926,
> 
>             "digest": "sha256:6a8487644ee8e5d4aac774380695b4dfc986a621a54f3d7a2f98c1dc6ad4a3a5",
> 
>             "platform": {
> 
>                 "architecture": "arm64",
> 
>                 "os": "linux"
> 
>             }
> 
>         },
> 
>         {
> 
>             "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
> 
>             "size": 595,
> 
>             "digest": "sha256:f711512233a66104534cf47880c2426dd9796b63ff12d733b0a94c6064b1cfb1",
> 
>             "platform": {
> 
>                 "architecture": "amd64",
> 
>                 "os": "linux"
> 
>             }
> 
>         }
> 
>     ]
> 
> }
> 
> ```
> 
> 
> 
> But when I go the the images page, I have this:
> 
> 
> 
> image
> 
> 
> 
> 
> 
> 

What's the exact cli command so I can try and replicate?

**Comment by cdrage** [comment]
> Can't make it to work.
> 
> I create a manifest using CLI:
> 
> ```
> 
> 
> 
> {
> 
>     "schemaVersion": 2,
> 
>     "mediaType": "application/vnd.oci.image.index.v1+json",
> 
>     "manifests": [
> 
>         {
> 
>             "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
> 
>             "size": 926,
> 
>             "digest": "sha256:6a8487644ee8e5d4aac774380695b4dfc986a621a54f3d7a2f98c1dc6ad4a3a5",
> 
>             "platform": {
> 
>                 "architecture": "arm64",
> 
>                 "os": "linux"
> 
>             }
> 
>         },
> 
>         {
> 
>             "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
> 
>             "size": 595,
> 
>             "digest": "sha256:f711512233a66104534cf47880c2426dd9796b63ff12d733b0a94c6064b1cfb1",
> 
>             "platform": {
> 
>                 "architecture": "amd64",
> 
>                 "os": "linux"
> 
>             }
> 
>         }
> 
>     ]
> 
> }
> 
> ```
> 
> 
> 
> But when I go the the images page, I have this:
> 
> 
> 
> image
> 
> 
> 
> 
> 
> 

Also, what's up with the dots in your image list? Is an icon not loading correctly?

**Comment by cdrage** [comment]
@jeffmaury I have replied in https://github.com/containers/podman-desktop/issues/7606 but I am unable to replicate the manifests not appearing.

File: packages/renderer/src/lib/image/ImageColumnActions.spec.ts [hunk 1]
```diff
@@ -40,6 +40,7 @@ const image: ImageInfoUI = {
   status: 'UNUSED',
   icon: undefined,
   badges: [],
+  digest: 'sha256:1234567890',
```
Comment (deboer-tim) [inline packages/renderer/src/lib/image/ImageColumnActions.spec.ts]: Curious why digest is added here when looks optional (and optional handled)? Would prefer not add it here (and other tests) if not required.

**Review by deboer-tim — APPROVED** [review]
Reviewed this and forgot to enter a review a while ago (sorry), but this worked well in testing both times. It was super useful to see manifests correctly for the first time, and visually so much better as a user.

File: packages/renderer/src/lib/image/ImageColumnActions.spec.ts (hunk 1 above)
Comment (cdrage) [reply packages/renderer/src/lib/image/ImageColumnActions.spec.ts]: You are right, not needed!

**Comment by jeffmaury** [comment]
Can't make it to work (maybe Windows related):

My steps:

1. Create a Containerfile with `FROM quay.io/ai-lab/llamacpp_python`
2. Build it with AMD64 and ARM64
3. Images list shows the manifest but not the underlying images

**Comment by cdrage** [comment]
> Can't make it to work (maybe Windows related):
> 
> My steps:
> 
> 1. Create a Containerfile with `FROM quay.io/ai-lab/llamacpp_python`
> 2. Build it with AMD64 and ARM64
> 3. Images list shows the manifest but not the underlying images

So I found why it wasn't working for you and it's an intentional feature of podman / already in the CLI, I've replied in here: https://github.com/containers/podman-desktop/issues/7606

**Review by jeffmaury — APPROVED** [review]
LGTM

**Pull request merged**